### DISCUSSION TITLE
v2.3.0 (Beta)

### DISCUSSION THREAD

**Discussion by osy** [utmapp/UTM#3174] [Releases]
## Installation
Visit [https://getutm.app/install/][1] for the most up to date installation instructions.

## New Features (Summary)
* **URL automation and VM downloading** (#2670) (thanks @conath) Allows for controlling and downloading VMs with `utm://` URL. Check the [Wiki page](https://github.com/utmapp/UTM/wiki/URL-Scheme-–-Automation-for-UTM-app-&-VMs) for more information.
* (macOS) Improved cursor/keyboard capturing and handling: permit macOS shortcuts (minimize, full-screen) when not captured, fix various cursor bugs, basic support for VoiceOver and capturing, and more.
* (macOS) Various UI enhancements: export log/args to file (#3056), title bar changes (#3125), exit confirmation works beyond Cmd+Q/W (#3154) (thanks @ktprograms and @conath)

## Changes
* Fixed DNS resolution on IPv6
* Updated zh-Hans localization (thanks @ty-yqs)
* Ensure temporary files are deleted when VM creation is canceled (#3012)
* Support re-generating and customizing MAC address (#2724)
* On PC/Q35 devices, USB mouse and keyboard will be disabled in order to prevent interference with the PS/2 keyboard and mouse that is always added to the machines. Some software such as Windows Installer will not work with the second mice.
* (macOS) Move VM name in main window from title to subtitle, reduces confusion when switching windows (#3099) (thanks @conath)
* (macOS) Show confirmation alert when closing/quitting with the menu or the button (#3103) (thanks @ktprograms)
* (macOS) Error alert (instead of crash) when trying to use unsupported features on < macOS 11.3 (#3118)
* (macOS) Support starting input capture with Ctrl+Opt (#3119)
* (macOS) Reduced UI thread blocking in terminal mode with lots of output spewing (#2404, #2555)
* (macOS) Add alternative hotkey for capture (Cmd+Opt) (#2710)
* (macOS) Unhide cursor when popup alert shows (#3101)

## Known Issues
* Networking can be broken in some instances, we are still investigating the cause. (#3051, #3094)

Please check the full list on [Github][2] and help report any bug you find that is not listed.

[1]: https://getutm.app/install/
[2]: https://github.com/utmapp/UTM/issues

<hr /><em>This discussion was created from the release <a href='https://github.com/utmapp/UTM/releases/tag/v2.3.0'>v2.3.0 (Beta)</a>.</em>

**Comment by conath** [comment]
> On PC/Q35 devices, USB mouse and keyboard will be disabled in order to prevent interference with the PS/2 keyboard and mouse that is always added to the machines. Some software such as Windows Installer will not work with the second mice.

The `usb-tablet` is still generated, right?

**Reply by osy** [reply]
Yes

**Comment by ktprograms** [comment]
What's the fullscreen shortcut?

Also Command+Option doesn't work to release the mouse.

**Reply by osy** [reply]
Full screen shortcut is indicated in the View menu. Cmd+Opt can be enabled in Preferences.

**Reply by ktprograms** [reply]
There doesn't seem to be a shortcut assigned:

<img width="648" alt="Screenshot 2021-10-11 at 8 55 47 AM" src="https://user-images.githubusercontent.com/71804605/136719522-e2f61599-d483-40e6-b38a-6cf7dd8af698.png">

**Reply by osy** [reply]
On my Monterey machine it’s fn+F but I guess maybe SwiftUI doesn’t automatically assign it. I’ll look into it but unfortunately I don’t have access to Big Sur right now.

**Reply by osy** [reply]
If you create a brand new empty SwiftUI project for mac, do you see the shortcut for enter full screen?

**Reply by ktprograms** [reply]
No

<img width="228" alt="Screenshot 2021-10-11 at 9 41 45 AM" src="https://user-images.githubusercontent.com/71804605/136721758-04f8105d-9c57-49a5-9e5c-15f04309e4f6.png">

**Reply by osy** [reply]
Well that’s an annoying bug. I can manually add “enter full screen” but then there will be two entries as there doesn’t seem to be a way to remove it.

**Reply by ktprograms** [reply]
Should there be just a keyboard shortcut without another menu item? Is that possible?

**Reply by osy** [reply]
Sure it can be hacked in but then it doesn’t follow Apple’s own UI guidelines lol. Probably will be the only way.

**Comment by conath** [comment]
I've added URL Scheme documentation to the [Wiki](https://github.com/utmapp/UTM/wiki/URL-Scheme-–-Automation-for-UTM-app-&-VMs). Edit: took the liberty to add the link to the above discussion post.

**Comment by sanberkzulfikar** [comment]
https://getutm.app/install/ link only mentions iOS devices. 

How can I update my current UTM version (2.2.4) and existing VMs to the UTM 2.3.0beta ?

**Reply by conath** [reply]
Go to the [Release page](https://github.com/utmapp/UTM/releases/tag/v2.3.0) and Download the .dmg File, then drag the contained .app File to the Applications folder and Replace.
